tests/lax_numpy_test.py

```diff
@@ -1752,6 +1752,27 @@ def testTrimZeros(self, a_shape, dtype, trim):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=True)
 
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_shape={}_rank{}".format(
+        jtu.format_shape_dtype_string(a_shape, dtype), rank),
+       "dtype": dtype, "a_shape": a_shape, "rank": rank}
+      for rank in (1, 2)
+      for dtype in default_dtypes
+      for a_shape in one_dim_array_shapes))
+  def testPoly(self, a_shape, dtype, rank):
+    if np.mintypecode(dtype(0).dtype.char) == 'd' and not config.x64_enabled:
+      self.skipTest(f"Only run {dtype} testcase when float64 is enabled.")
+    elif dtype == np.float32 and not config.x64_enabled and rank == 2:
+      self.skipTest(f"Only run {dtype} testcase with rank 2 when "
+                    "float64 is enabled (due to the need for complex64).")
+    elif dtype == np.float16:
+      self.skipTest(f"{dtype} not supported in numpy.")
```

Comment:
Perhaps skip `CheckAgainstNumpy`, but still run `CompileAndCheck`?

Reply:
It seems that we cannot run `CompileAndCheck` since `jnp.linalg.eigvals` is currently not implemented for float16. 
```

>>> jnp.linalg.eigvals(jnp.array([[1,2],[3,4]], dtype=jnp.float16))
a DeviceArray([[1., 2.],
             [3., 4.]], dtype=float16)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/pnguyen/code/jax/jax/_src/numpy/linalg.py", line 294, in eigvals
    compute_right_eigenvectors=False)[0]
  File "/Users/pnguyen/code/jax/jax/_src/lax/linalg.py", line 82, in eig
    return eig_p.bind(x, compute_left_eigenvectors=compute_left_eigenvectors,
  File "/Users/pnguyen/code/jax/jax/core.py", line 264, in bind
    out = top_trace.process_primitive(self, tracers, params)
  File "/Users/pnguyen/code/jax/jax/core.py", line 603, in process_primitive
    return primitive.impl(*tracers, **params)
  File "/Users/pnguyen/code/jax/jax/_src/lax/linalg.py", line 374, in eig_impl
    xla.apply_primitive(eig_p, operand,
  File "/Users/pnguyen/code/jax/jax/interpreters/xla.py", line 234, in apply_primitive
    compiled_fun = xla_primitive_callable(prim, *unsafe_map(arg_spec, args), **params)
  File "/Users/pnguyen/code/jax/jax/_src/util.py", line 186, in wrapper
    return cached(config._trace_context(), *args, **kwargs)
  File "/Users/pnguyen/code/jax/jax/_src/util.py", line 179, in cached
    return f(*args, **kwargs)
  File "/Users/pnguyen/code/jax/jax/interpreters/xla.py", line 276, in xla_primitive_callable
    built_c = primitive_computation(prim, AxisEnv(nreps, (), ()), backend,
  File "/Users/pnguyen/code/jax/jax/_src/util.py", line 186, in wrapper
    return cached(config._trace_context(), *args, **kwargs)
  File "/Users/pnguyen/code/jax/jax/_src/util.py", line 179, in cached
    return f(*args, **kwargs)
  File "/Users/pnguyen/code/jax/jax/interpreters/xla.py", line 317, in primitive_computation
    ans = rule(c, *xla_args, **params)
  File "/Users/pnguyen/code/jax/jax/_src/lax/linalg.py", line 414, in eig_cpu_translation_rule
    w, vl, vr, info = _cpu_geev(c, operand, jobvl=compute_left_eigenvectors,
  File "jaxlib/lapack.pyx", line 1785, in lapack.geev
    raise NotImplementedError("Unsupported dtype {}".format(dtype))
NotImplementedError: Unsupported dtype float16
>>> 
```
Since this is a `NotImplementedError`, should I just skip `CheckAgainstNumpy` and call`pytest.xfail` right before running `CompileAndCheck` when the dtype is float16?

Reply:
If float16 is not compatible with either compilation or numpy, it's probably fine to skip the test (or ensure that float16 is left out of the parametrization above)

Reply:
That makes sense. I simply skip the test in https://github.com/google/jax/commit/d3c3cfb43eeaa9bf30944b6550be0097ceefafd5.